src/conversion.rs

```diff
@@ -1756,6 +1772,10 @@ mod tests {
         assert!(u32::try_from(u39(0x1_fffff_ffff)).is_err());
 
         assert!(u6::try_from(u7(127)).is_err());
+
+        assert_eq!(u2::try_from(1usize), Ok(u2(1)));
+        assert!(u2::try_from(4usize).is_err());
+        assert_eq!(u2(1).try_into(), Ok(1usize));
```

Comment:
I feel like we should have some test cases for the failing case from uX to usize for completing all cases. Would you mind add it? To make it portable we can make it as a few different tests that asserts that the behavior is identical as for a core type conversion? Something like below

```rust
assert_eq!(usize::try_from(0x1_ffffu64).is_err(), usize::try_from(u42(0x1_ffff)).is_err())
assert_eq!(usize::try_from(0x1_ffff_ffffu128).is_err(), usize::try_from(u68(0x1_ffff_ffff)).is_err())
```

Do you agree that this makes sense?

Reply:
^It should probably be accompanied with a comment to describe that the reason we test it this way is due to portability between 16, 32 and 64 bit architectures

Reply:
> I feel like we should have some test cases for the failing case from uX to usize for completing all cases. Would you mind add it? To make it portable we can make it as a few different tests that asserts that the behavior is identical as for a core type conversion? Something like below
> 
> ```rust
> assert_eq!(usize::try_from(0x1_ffffu64).is_err(), usize::try_from(u42(0x1_ffff)).is_err())
> assert_eq!(usize::try_from(0x1_ffff_ffffu128).is_err(), usize::try_from(u68(0x1_ffff_ffff)).is_err())
> ```
> 
> Do you agree that this makes sense?

Yup, this is a good idea. Will add that and a comment.

Reply:
After reading this again, I think I'm missing the significant of using `u42` and `u68` specifically.  The intention here is check something like: 

* "on a 16-bit arch, a 17 bit value should fail to convert to usize, so trying to convert a u32 and a u17 should have the same result"
* "on a 32-bit arch, a 33 bit value should value to convert to usize, so trying to convert a u64 and a u33 should have the same result"
* "on a 64-bit arch, a 65 bit value should value to convert to usize, so trying to convert a u128 and a u65 should have the same result"

Right?  Or perhaps I'm misunderstanding you.  I will say that I only implemented usize conversions up to u63, mainly because that's where all the other conversions seemed to "stop".

Reply:
The only reason to use `u42` and `u68` is that they are large enough containers to test failures on systems where `usize` is 32 and 64 bit. 

> "on a 16-bit arch, a 17 bit value should fail to convert to usize, so trying to convert a u32 (the next-largest built-in type) and a u17 (the next-largest uX type) should have the same result"

The number that is being converted is the exact same number. The ability to convert into `usize` is only dependent on the value and not the type. This means `0x1_ffffu64` and `u42(0x1_ffff)` should either both succeed or fail not matter what platform the test is being run on. The reason for using a value larger than `u64::MAX`  was to trigger an error on most (all?) platform. Since the conversions are not implemented we can wait with this test until they are.

Reply:
> This means 0x1_ffffu64 and u42(0x1_ffff) should either both succeed or fail not matter what platform the test is being run on

Sorry for being dense, but I think this is where you're losing me.  0x1_ffff takes up 17 bits, how does that validate uX type conversions into usize match whatever u32/u64 does?  Shouldn't it be:

```
// No matter the arch, a u33 (the first uX type that spills into a u64) with value u33::MAX should have the same result as a u64 with value u33::MAX (`0x1_FFFF_FFFF`)
assert_eq!(usize::try_from(0x1_FFFF_FFFFu64).is_err(), usize::try_from(u33::MAX).is_err())
// No matter the arch, a u65 (the first uX type that spills into a u128) with value u65::MAX should have the same result as a u128 with a value of u65::MAX (`0x1_FFFF_FFFF_FFFF_FFFF`)
assert_eq!(usize::try_from(0x1_FFFF_FFFF_FFFF_FFFF).is_err(), usize::try_from(u65::MAX).is_err())
```

Reply:
Yes, you are absolutely right.

Reply:
Since the conversions are not supported over for u65 I think it's also fine to not test it and only add the one for `u33`. If you decide to add for `u65` that is also great, but then it would be awesome if you could also add the rest of the methods\conversions to make `u65` actually useful instead of only a detail for testing this very specific thing that will be difficult to use practice.

Both are completely fine with me, your choice!

Reply:
I got rid of the u65 test case, maybe another time I'll add all the conversions for the other types :).  Believe I got the formatting issues fixed as well.